torch/profiler/utils.py

```diff
@@ -0,0 +1,116 @@
+from collections import deque
+from dataclasses import dataclass
+from torch.profiler import profile, DeviceType
+import torch
```

Comment:
You shouldn't use `import torch` inside framework code, because if we decide to expose symbols in the torch namespace it leads to a circular import.